File: src/main/java/arriba/fix/disruptor/FixMessageEntry.java
package arriba.fix.disruptor;

import arriba.fix.messages.FixMessage;
import com.lmax.disruptor.AbstractEvent;
import org.jboss.netty.buffer.ChannelBuffer;

public final class FixMessageEntry extends AbstractEvent {

    private ChannelBuffer serializedFixMessage;
    private FixMessage fixMessage;

    public FixMessageEntry() {}

    public void setFixMessage(final FixMessage fixMessage) {
        this.fixMessage = fixMessage;
    }

    public FixMessage getFixMessage() {
        return this.fixMessage;
    }

    public ChannelBuffer getSerializedFixMessage() {
        return this.serializedFixMessage;
    }

    public void setSerializedFixMessage(final ChannelBuffer serializedFixMessage) {
        this.serializedFixMessage = serializedFixMessage;
    }
}

File: src/main/scala/arriba/scala/fix/disruptor/FixMessageEntry.scala
package arriba.scala.fix.disruptor

import com.lmax.disruptor.AbstractEntry
import arriba.scala.fix.messages.FixMessage
import arriba.scala.fix.SerializedField

class FixMessageEntry extends AbstractEntry {

  def setFixMessage(fixMessage: FixMessage) {
    this.fixMessage = fixMessage
  }

  def getFixMessage: FixMessage = {
    this.fixMessage
  }

  def setSerializedFields(serializedFields: List[SerializedField]) {
    this.serializedFields = serializedFields
  }

  def getSerializedFields: List[SerializedField] = {
    this.serializedFields
  }

  private var serializedFields: List[SerializedField] = null
  private var fixMessage: FixMessage = null
}

